src/d3a/models/strategy/infinite_bus.py

```diff
@@ -57,17 +60,27 @@ def _set_market_maker_rate(self):
                 ConstSettings.GeneralSettings.DEFAULT_MARKET_MAKER_RATE
 
     def event_activate(self, **kwargs):
-        if self.energy_rate_profile is not None:
+        if self.energy_rate_profile is not None and isinstance(self.energy_rate_profile, dict):
             self.energy_rate = read_and_convert_identity_profile_to_float(self.energy_rate_profile)
             del self.energy_rate_profile
+        elif self.energy_rate_profile is not None and isinstance(self.energy_rate_profile, str):
+            self.energy_rate = read_arbitrary_profile(InputProfileTypes.IDENTITY,
+                                                      self.energy_rate_profile)
+            del self.energy_rate_profile
         else:
             self.energy_rate = self.area.config.market_maker_rate if self.energy_rate is None \
                 else read_arbitrary_profile(InputProfileTypes.IDENTITY, self.energy_rate)
 
-        if self.buying_rate_profile is not None:
+        if self.buying_rate_profile is not None and \
+                isinstance(self.buying_rate_profile, (float, dict)):
             self.energy_buy_rate = \
                 read_and_convert_identity_profile_to_float(self.buying_rate_profile)
             del self.buying_rate_profile
+        elif self.buying_rate_profile is not None and isinstance(self.buying_rate_profile, str):
+            self.energy_buy_rate = \
+                read_arbitrary_profile(InputProfileTypes.IDENTITY, self.buying_rate_profile)
+            del self.buying_rate_profile
+
```

Comment:
If its an `str` it contains file path to csv where profile is for local run, and in other case it could be ~float~`dict` being passed from d3a-web.

Reply:
I think that `read_arbitrary_profile` already handles all these cases. Please make sure you double-check so that we could possibly simplify these `if` statements :)